### PR TITLE
fix(sdk): maybe add proof node

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Fixes a bug in the SDK that would cause it to be unable to prove certain types of withdrawals. Bug can be easily fixed on the client side by adding an extra proof element in certain cases.

**Tests**

Adds a test for the case where a node is added and for the case where a node does not need to be added.

### CODE REVIEW

**Review by roninjin10 — COMMENTED** [review]
This is going over my head what bug this pr is fixing. Can you ELI5?

**Review by tynes — APPROVED** [review]

**Review by roninjin10 — APPROVED** [review]
